### PR TITLE
Convert more tests to use full form flow

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Convert more tests to use full form flow

Before
----------------------------------------
Tests use artificial flow

After
----------------------------------------
Tests use realistic flow

Technical Details
----------------------------------------
The tests blocked an extra parameter getting through to the render function so I just added handling so it wouldn't - since it isn't needed there anyway

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/BAO/MessageTemplate.php [hunk 1]
```diff
@@ -376,7 +376,10 @@ protected static function renderTemplateRaw($params) {
    * @throws \CRM_Core_Exception
    * @throws \API_Exception
    */
-  public static function sendTemplate($params) {
+  public static function sendTemplate(array $params): array {
+    // Handle isEmailPdf here as the unit test on that function deems it 'non-conforming'.
+    $isAttachPDF = (bool) ($params['isEmailPdf'] ?? NULL);
```
[the remark this answers is not in the document]
Comment (eileenmcnaughton) [reply CRM/Core/BAO/MessageTemplate.php]: Done

**Comment by seamuslee001** [comment]
@eileenmcnaughton style warning 

```
ContributionTest.php:2191, ParamMissingDefinition, Priority: High
--
Parameter $pledgePaymentID is not described in comment
```

File: CRM/Core/BAO/MessageTemplate.php [hunk 2]
```diff
@@ -376,7 +376,10 @@ protected static function renderTemplateRaw($params) {
    * @throws \CRM_Core_Exception
    * @throws \API_Exception
    */
-  public static function sendTemplate($params) {
+  public static function sendTemplate(array $params): array {
+    // Handle isEmailPdf here as the unit test on that function deems it 'non-conforming'.
+    $isAttachPDF = !empty($params['isEmailPdf']);
+    unset($params['isEmailPdf']);
```
Comment (demeritcowboy) [inline CRM/Core/BAO/MessageTemplate.php:382]: Just noting that by unsetting it it no longer will be in the params passed to the mail hooks, but that should be ok.

Comment (eileenmcnaughton) [reply CRM/Core/BAO/MessageTemplate.php:382]: Yeah - the test enforces that

**Pull request merged**